app/routes/events/utils.js

```diff
@@ -164,6 +163,10 @@ export const transformEvent = (data: TransformEvent) => ({
   priceMember: calculatePrice(data),
   location: calculateLocation(data),
   paymentDueDate: calculatePaymentDueDate(data),
+  canViewGroups: data.isGroupOnly
+    ? data.canViewGroups.map((group) => group.id)
+    : [],
+  requireAuth: data.canViewGroups.length > 0,
```

Comment:
Not a big deal for this PR, but for the future we should discuss having this option separate from the `canviewGroups` as it does provide some different permission stuff.